sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4140,6 +4140,20 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-34581: Don't optimize out grouping expressions from aggregate expressions") {
+    withTempView("t") {
+      Seq[Integer](null, 1, 2, 3, null).toDF("id").createOrReplaceTempView("t")
+
+      val df = spark.sql(
```

Comment:
If this is a test to check if a query work correctly or not, could you move this test into `SQLQueryTestSuite`?

Reply:
It is, moved in https://github.com/apache/spark/pull/31913/commits/09f1a85d8312e56c886b31f26ae87d245288d2b9